prime-dotnet-webapi/Services/OrganizationService.cs

```diff
@@ -64,6 +67,15 @@ public async Task<IEnumerable<OrganizationSearchViewModel>> GetOrganizationsAsyn
                 .DecompileAsync()
                 .ToListAsync();
 
+            // TODO: Fix "A second operation started on this context before a previous operation completed." error
```

Comment:
Is this error still occurring?

Reply:
No the error doesn't occur with the code as-is but I have a feeling the code could be improved.

Reply:
If you change this back to 
```C#
var result = await _context.Organizations
``` 
does the error return? Or convention is to await `Task`s rather than do task.Result.

Reply:
I reverted line 55 and `/adjudication/site-registrations` loads fine.  But if I have the following code enabled, I get that error again.  Assuming it's related to https://docs.microsoft.com/en-us/ef/core/dbcontext-configuration/#asynchronous-operation-pitfalls but not sure how to correct it without using `task.Result`   

![image](https://user-images.githubusercontent.com/7586665/125485639-d7dba6a6-8b82-4f3a-b716-cb79edc085cb.png)

Reply:
The error `A second operation started on this context before a previous operation completed.` suggests that multiple DB connections are being opened simultaneously. I strongly suspect the issue was the line 
```C#
results.ForEach(async r => r.IsUnderReview = await organizationClaimService.IsOrganizationUnderReviewAsync(r.Id));
```
which is opening a db connection for each organization being returned. The better way to do this is to add the mapping to the automapper configuration so the Property gets filled out when you do `.ProjectTo<OrganizationListViewModel>`, rather than doing another DB call for each item.

Reply:
There are a few ways you could do this, with minor tradeoffs between them. One way is to directly supply the `OrganizationClaims` table to Automapper:
```C#
// In AutoMapping.cs
IQueryable<OrganizationClaim> claims = null;
CreateMap<Organization, OrganizationListViewModel>()
    .ForMember(dest => dest.HasAcceptedAgreement, opt => opt.MapFrom(src => src.Agreements.Any(a => a.AcceptedDate.HasValue)))
    .ForMember(dest => dest.HasSubmittedSite, opt => opt.MapFrom(src => src.Sites.Any(s => s.SubmittedDate.HasValue)))
    .ForMember(dest => dest.Sites, opt => opt.MapFrom(src => src.Sites.Where(s => careSettingCode == null || s.CareSettingCode == careSettingCode)))
    .ForMember(dest => dest.HasClaim, opt => opt.MapFrom(src => claims.Any(c => c.OrganizationId == src.Id)));

// In OrganizationService.cs, line 66
.ProjectTo<OrganizationListViewModel>(_mapper.ConfigurationProvider, new { careSettingCode = searchOptions.CareSettingCode, claims = _context.OrganizationClaims })
```
This has the upside of not having to change any Models or View Models etc at the cost of making the mapping / fetching code a little uglier.

Another option would be to add the `Claims` navigation property to `Organization`:
```C#
// In AutoMapping.cs
CreateMap<Organization, OrganizationListViewModel>()
    .ForMember(dest => dest.HasAcceptedAgreement, opt => opt.MapFrom(src => src.Agreements.Any(a => a.AcceptedDate.HasValue)))
    .ForMember(dest => dest.HasSubmittedSite, opt => opt.MapFrom(src => src.Sites.Any(s => s.SubmittedDate.HasValue)))
    .ForMember(dest => dest.Sites, opt => opt.MapFrom(src => src.Sites.Where(s => careSettingCode == null || s.CareSettingCode == careSettingCode)))
    .ForMember(dest => dest.HasClaim, opt => opt.MapFrom(src => src.Claims.Any()));
    
// In Organization.cs
[JsonIgnore]
public ICollection<OrganizationClaim> Claims { get; set; }
```
This has the upside of being the simplest code change with the downside of adding more stuff to the `Organization` Model.

another option would be to take that last one and expand on it:
```C#
// In Organization.cs
[NotMapped]
[Computed]
public bool HasClaim
{
    get => Claims.Any();
}
```
This has the upside of not needing any change to Automapper at all, and it solves the problem of needing the property on both the `OrganizationListViewModel` and `Organization` itself (since we still return the whole `Organization` on GET organizations/{id} rather than a View Model). This has the downside of having *another* not mapped, computed property on our models. I would expect this would go away when we get a proper `OrganizationViewModel`.

My preference would be either option 2 or 3, fairly equally.

Reply:
@james-hollinger , thanks for the ideas.  It became something like Options 2 and 3.  Please have a look.